linera-chain/src/block.rs

```diff
@@ -109,3 +112,21 @@ impl From<Timeout> for HashedCertificateValue {
         HashedCertificateValue::new_timeout(value.chain_id, value.height, value.epoch)
     }
 }
+
+impl From<HashedCertificateValue> for Hashed<Timeout> {
+    fn from(hv: HashedCertificateValue) -> Hashed<Timeout> {
+        let hash = hv.hash();
+        match hv.into_inner() {
+            CertificateValue::Timeout(timeout) => Hashed::unchecked_new(timeout, hash),
+            _ => panic!("Expected a Timeout value"),
```

Comment:
I would feel more confident if we used `TryFrom` instead of `From`, and then `expect`ed or `unwrap`ped at the call site if we're sure.

Reply:
I'll change this to `TryFrom` and make a separate PR where I'll make a sweep over other ones

Reply:
Done in [Replace From with TryFrom](https://github.com/linera-io/linera-protocol/pull/2887/commits/bdbad2f527b1de1ac3a773dee42ccd6d40577289)